trillian/migrillian/main.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"google.golang.org/grpc"
+
+	"github.com/google/certificate-transparency-go/client"
+	"github.com/google/certificate-transparency-go/jsonclient"
+	"github.com/google/certificate-transparency-go/scanner"
+	"github.com/google/trillian"
+)
+
+var (
+	ctLogURI    = flag.String("ct_log_uri", "http://ct.googleapis.com/aviator", "CT log base URI to fetch entries from")
+	trillianURI = flag.String("trillian_uri", "localhost:8091", "Trillian log server URI to add entries to")
+	logID       = flag.Int64("log_id", 0, "Trillian log tree ID to add entries to")
+
+	batchSize      = flag.Int("batch_size", 512, "Max number of entries to request per get-entries call")
+	parallelFetch  = flag.Int("parallel_fetch", 2, "Number of concurrent get-entries fetchers")
+	parallelSubmit = flag.Int("parallel_submit", 2, "Number of concurrent AddSequencedLeaves submitters")
+
+	startIndex = flag.Int64("start_index", 0, "CT log index to start scanning at")
+	endIndex   = flag.Int64("end_index", 0, "CT log index to end scanning at (non-inclusive, 0 = end of log)")
+
+	quiet = flag.Bool("quiet", false, "Don't print out extra logging messages")
```

Comment:
This is the opposite of most of our other binaries? Not a huge issue probably.

Reply:
This was initially copied from `scanlog`.
Are you suggesting to make this `true` by default? I am very much for it.

Reply:
I meant that mostly we often -verbose flags and it's quiet by default, though it's not a fixed rule.